packages/synapse-react-client/src/utils/functions/queryUtils.ts

```diff
@@ -241,3 +245,40 @@ export function removeEmptyQueryParams(q: Query) {
 
   return query
 }
+
+/**
+ * Given a FacetColumnResult and a set of ColumnModels, return the ColumnModel that
+ * matches the FacetColumnResult.
+ * @param facet
+ * @param columnModels
+ */
+export function getCorrespondingColumnForFacet(
+  facet: FacetColumnResult,
+  columnModels: ColumnModel[],
+): ColumnModel | JsonSubColumnModel | undefined {
+  let columnModel: ColumnModel | JsonSubColumnModel | undefined =
+    columnModels.find(model => model.name === facet.columnName)
+  if (facet.jsonPath && columnModel && columnModel.jsonSubColumns) {
+    columnModel = columnModel.jsonSubColumns.find(
+      cm => cm.jsonPath === facet.jsonPath,
+    )
+  }
+  return columnModel
+}
```

Comment:
In a later PR I refactored the facet controls. I extracted and added tests for this common functionality